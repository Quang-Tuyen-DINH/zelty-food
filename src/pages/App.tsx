import React from "react";
import { ToastContainer } from "react-toastify";
import { AppStyled } from "../components/styles/App.styled";
import AppRoutes from "../routes/AppRoutes";

function App() {

  return (
    <AppStyled className="zelty-restaurant">
      <AppRoutes />
      <ToastContainer />
    </AppStyled>
  );
}

export default App;
