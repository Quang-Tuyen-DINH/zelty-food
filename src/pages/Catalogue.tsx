import React, { useEffect, useState } from "react";
import { MenuList } from "../components/ui/Menu";
import { SearchInput } from "../components/forms/SearchInput";
import { CatalogueStyled } from "../components/styles/Catalogue.styled";
import Cart from "../components/ui/Cart";
import CatalogueService from "../services/catalogue/Catalogue.service";
import { useDispatch } from "react-redux";
import Store from "../store/Index";
import { Menu } from "../shared/models/Menu.model";
import { Product } from "../shared/models/Product.model";
import { Item } from "../shared/models/Option.model";
import ProductCard from "../components/ui/ProductCard";

export const Catalogue = () => {
  const dispatch = useDispatch();
  const [products, setProducts] = useState<Product[]>([]);
  const [options, setOptions] = useState<Item[]>([]);
  const [menu, setMenu] = useState<Menu[]>([]);
  const [selectedMenu, setSelectedMenu] = useState<string>("M01")
  const [searching, setSearching] = useState<boolean>(false);
  const [keyword, setKeyword] = useState<string>("")

  useEffect(() => {
    getMenu();
    getAllProducts();
    getAllOptions();
    Store.subscribe(() => {
      setMenu(Store.getState().menu);
      setProducts(Store.getState().products);
      setOptions(Store.getState().options);
    });
  }, [])

  const getMenu = async () => {
    await CatalogueService.getMenu()
      .then((data) => {
        if(!data.data.Items) {
          return;
        }
        dispatch({ type: "LOAD_MENU", payload: data.data.Items });
      })
  }

  const selectMenu = (menuId: string) => {
    setSelectedMenu(menuId);
    setKeyword("");
    setSearching(false);
  }

  const getAllProducts = async () => {
    await CatalogueService.getAllProducts()
      .then((data) => {
        if(!data.data.Items) {
          return;
        }
        dispatch({ type: "LOAD_PRODUCTS", payload: data.data.Items });
      })
      .catch(error => {console.log(error)})
  }

  const getAllOptions = async () => {
    await CatalogueService.getAllOptions()
      .then((data) => {
        if(!data.data.Items) {
          return;
        }
        const options = [].concat(
          ...data.data.Items.map((group: any) => {
              return group.items;
            }
          )
        )
        dispatch({ type: "LOAD_OPTIONS", payload: options });
      })
      .catch(error => {console.log(error)})
  }

  const searchProduct = (event: any) => {
    if(event.target.value.length > 0) {
      setSearching(true);
      setKeyword(event.target.value);
    } else {
      setSearching(false);
    }
  }

  return (
    <CatalogueStyled className="zelty-restaurant__catalogue">
      <div className="zelty-restaurant__catalogue__left">
        <SearchInput searchProduct={searchProduct}/>
        <MenuList menu={menu} selectMenu={selectMenu}/>
        <div className="zelty-restaurant__catalogue__left__products">
          {searching === false && products.filter((product: Product) => product.menuId === selectedMenu).map((product: Product) => (
            product.available_options ?
              <ProductCard key={product.id} product={product} options={options}/>
            :
              <ProductCard key={product.id} product={product} />
          ))}
          {searching === true && products.filter((product: Product) => product.name.toLowerCase().includes(keyword.toLowerCase())).map((product: Product) => (
            product.available_options ?
              <ProductCard key={product.id} product={product} options={options}/>
            :
              <ProductCard key={product.id} product={product} />
          ))}
        </div>
      </div>
      <div className="zelty-restaurant__catalogue__right">
        <Cart atCatalogue={true} atCheckout={false}  checkoutConfirmed={false}/>
      </div>
    </CatalogueStyled>
  );
};
