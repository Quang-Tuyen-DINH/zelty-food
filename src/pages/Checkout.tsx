import { CheckoutInput } from "../components/forms/CheckoutInput";
import { CheckoutStyled } from "../components/styles/Checkout.styled";

export const Checkout = () => {
  return (
    <CheckoutStyled className="zelty-restaurant__checkout">
      <div className="zelty-restaurant__products">
        <CheckoutInput />
        CHECKOUT
      </div>
    </CheckoutStyled>
  );
};
