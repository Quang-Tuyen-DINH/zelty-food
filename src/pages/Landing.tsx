import React from 'react';
import { useNavigate } from 'react-router-dom';
import { LandingStyled } from "../components/styles/Landing.styled";
import { Button } from '../components/ui/Button';

const Landing = () => {
  const navigate = useNavigate();

  const navigateCheckout = () => {
    navigate("/catalogue");

  }
  return (
    <LandingStyled className="zelty-restaurant__landing">
      <div className="zelty-restaurant__landing__title">Bienvenue sur <span>Zelty</span></div>
      <div className="zelty-restaurant__landing__commence">
        <Button className="zelty-restaurant__landing__commence__button" onClick={navigateCheckout}>Commencer Ã  commander</Button>
      </div>
    </LandingStyled>
  )
}

export default Landing