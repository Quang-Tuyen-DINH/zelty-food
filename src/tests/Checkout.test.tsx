import { render, screen, cleanup } from '@testing-library/react';
import { Provider } from 'react-redux'
import Adapter from "@cfaester/enzyme-adapter-react-18";
import { shallow, configure } from "enzyme";
import { Checkout } from '../pages/Checkout';
import Store from '../store/Index';

afterEach(cleanup);

configure({adapter: new Adapter()});

describe("Test Checkout", () => {
  it('tests the name', () => {
  
    const product = shallow(
      <Provider store={Store}>
        <Checkout />
      </Provider>
    ).dive()
  
    expect(product).toMatchSnapshot();
  });
})