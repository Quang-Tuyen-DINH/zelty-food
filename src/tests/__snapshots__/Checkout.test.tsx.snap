// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Checkout tests the name 1`] = `ShallowWrapper {}`;
